core/arch/arm/kernel/ree_fs_ta.c

```diff
@@ -247,7 +282,23 @@ static TEE_Result ree_fs_ta_read(struct user_ta_store_handle *h, void *data,
 		return TEE_ERROR_BAD_PARAMETERS;
 	if (data) {
 		dst = data; /* Hash secure buffer (shm might be modified) */
-		memcpy(dst, src, len);
+		if (handle->shdr->img_type == SHDR_ENCRYPTED_TA) {
+			/*
+			 * Encrypted TA loading only support buffered REE-FS
```

Comment:
Why does encrypted TA loading only support buffered REE-FS?

Reply:
Since it doesn't seems feasible to partially decrypt and read from encrypted TA image. So we need to read encrypted TA image at once and decrypt in temporary secure memory buffer.